activation/activation.go

```diff
@@ -190,8 +196,10 @@ func (b *Builder) StartSmeshing(coinbase types.Address, opts PostSetupOpts) erro
 			// ensure we are ATX synced before starting the PoST Session
 		}
 
-		if err := b.postSetupProvider.StartSession(ctx, opts); err != nil {
-			return err
+		// If start session returns any error other than context.Canceled
+		// (which is how we signal it to stop) then we panic.
+		if err := b.postSetupProvider.StartSession(ctx, cfg); err != nil && !errors.Is(err, context.Canceled) {
+			panic(fmt.Sprintf("initialization failed: %v", err))
```

Comment:
Why panic instead of returning the error? If this is caused by a GRPC request the error that caused it won't be propagated to smapp and other running go-routines will forcefully be shut down as well?

Reply:
It's not via API. If you have post data in the directory and you configure go-spacemesh /wrongly/ then it will silently exit currently.

We don't want that. ;) The correct fix is for later ;)

Reply:
I'm not a fan of `log.Fatal` since you lose the context of where in the stack your failure occurred, which in these critical failures can be pretty useful. However `log.Panic` seems like a sensible change, as we'll now log any extra contextual information that was added to the log.

Reply:
However actually when it comes to testing just using panic can be quite useful. Using `log.Fatal ` can be quite confusing if logging is not enabled for the test as you have previously discovered - https://spacemesh.slack.com/archives/C89QJJY3G/p1679421425885569

Reply:
The instance of `log` that is used to `Fatal` or `Panic` should already contain a lot of context that is necessary to debug an issue; the request ID, the service it was called in, relevant state of the service, etc.

I don't mind using `log.Panic` over `log.Fatal` but `panic` just prints text that might be more difficult to parse in Grafana/Loki.

Reply:
`log.Panic` it is then, although we don't seem to have any consistency around this point currently. I see a lot of plain calls to panic and also lots to `log.Panic` so maybe it would be worth trying to decide on an approach to use consistently.